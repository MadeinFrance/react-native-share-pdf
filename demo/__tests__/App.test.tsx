/**
 * @format
 */

import React from 'react';
import App from '../App';

import sharePDF from 'react-native-share-pdf';
import {fireEvent, render} from '@testing-library/react-native';

jest.mock('react-native-share-pdf', () => jest.fn());

describe('<App />', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<App />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should share the link', async () => {
    const {getByTestId, toJSON} = render(<App />);
    const shareButton = getByTestId('share-button');

    await fireEvent.press(shareButton);

    expect(sharePDF).toHaveBeenCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });
});
