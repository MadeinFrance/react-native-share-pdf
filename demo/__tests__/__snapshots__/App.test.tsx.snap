// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#222",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#222",
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#000",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Download PDF Base64
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="share-button"
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Share
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`<App /> should share the link 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#222",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#222",
      }
    }
  >
    <View>
      <View
        style={
          {
            "backgroundColor": "#000",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "margin": 10,
                "textAlign": "center",
              }
            }
          >
            Download PDF Base64
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="share-button"
          >
            <View
              style={
                [
                  {},
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                Share
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
